### PR TITLE
Site Creation: Domain page improvements

### LINKED ISSUE #8832: Site Creation: Suggested domains order and padding
I believe the suggested domains are ordered alphabetically and in most cases that means the exact query the user has typed is in the middle (or end) of the suggested domains instead of as the first result. From my very brief test, it looks like the result from the API request is ordered somewhat logically, but I haven't seen any code that's specifically ordering it either.

Since this is a lower priority issue, I am opening a separate issue for it, so we can take care of it before the launch of v1 of new site creation flow.

I've also found that the padding in the domains screen isn't working great when we have more than one line of text as shown in the screenshot below. @SylvesterWilmott Should we make a change to this?

screenshot_1545256364

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #8832. This is a very small PR that addresses a few minor issues in the domains page in the new site creation flow. Here are the issues addressed:

1. When the user types a site title in the site info page, we'll use that information to do an initial search so we can show some suggested domains to the user immediately. If that request fails, we used to show an error row which doesn't make too much sense for the user because they didn't initiate anything. I've updated the code to only show the error if it's from a user initiated query in b0c8a9a25615f47e7b531f66ae9da9576a454383. We still treat this as error, but we don't show it to the user which is different from the next fix.
2. While testing the site creation flow I found that typing `.` was returning `INVALID_QUERY` error which is not an error we need to specifically handle. I've introduced this error type in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1052 and updated the code to handle it in c73b9f0925ca4af22fa788e49aea9d3fcfecf712. We now treat the response as if it's a successful response and we couldn't find any domains for the current query. We could potentially change the message of the empty page, but I think this is good enough for now. Note that this is NOT treated as an error in the ViewModel which is different from the first fix.

_P.S: The difference between the two fixes is that first is still an error, but since it's not user initiated we don't want to bother the user. The second is not really an error and no different than an empty response. That's why they are handled in different places._

3. I've added a little bit padding to the top and bottom of the row so that multiline suggestions look a bit better. I've asked about this in #8832 to @SylvesterWilmott but he didn't have a padding in mind, so I am trying one that looks to be good and we'll update it if necessary during the design review in #8864. Below you can find two screenshots that show before and after versions for the padding change:

_Before:_
screenshot_1545865321

_After:_
screenshot_1545865971

As far as I can tell there is not a significant difference for single rows, so we should be fine for that 🤞 

---

**To test:**
_How to access site info page: Go to site picker, tap on the `+` menu button, select `Create WordPress.com site`, select any segment, skip verticals page (or fill in, doesn't matter)_

**For first fix:**
* Go to site info page and type `.` to the title
* Tap next and verify that after the request is finished (progressbar stops spinning) nothing happens :)

**For second fix:**
* Go to site info page and skip it to access domains page
* Type `.` and verify that it says `No available addresses matching your search` in the empty view

**For third fix:**
* Go to site info page and skip it to access domains page
* Type a very long query (enough for the results to be two row) and verify the padding is somewhat OK and similar to the `after` screenshot shared above. (we'll get a design review for this later)

---

A single reviewer should be enough for this PR.

### CODE REVIEW

File: WordPress/src/main/res/layout/new_site_creation_domains_item.xml [hunk 1]
```diff
@@ -18,7 +18,10 @@
         android:clickable="false"
         android:focusable="false"
         android:focusableInTouchMode="false"
+        android:paddingBottom="@dimen/margin_medium"
+        android:paddingEnd="@dimen/margin_large"
```
Comment (theck13) [inline WordPress/src/main/res/layout/new_site_creation_domains_item.xml]:
```suggestion
        android:paddingEnd="@dimen/margin_extra_large"
```

Shouldn't this be `margin_extra_large` for the end padding to match 16dp from @SylvesterWilmott's [comment](https://github.com/wordpress-mobile/WordPress-Android/issues/8832#issuecomment-449345909)?

Comment (oguzkocer) [reply WordPress/src/main/res/layout/new_site_creation_domains_item.xml]: I missed that comment, thanks for noticing @theck13!

Comment (oguzkocer) [reply WordPress/src/main/res/layout/new_site_creation_domains_item.xml]: When you apply the suggested commit it's authored by both, really like that! And thank you for the suggestion line as well @theck13, it saved me a bunch of time from switching branches :)

Comment (theck13) [reply WordPress/src/main/res/layout/new_site_creation_domains_item.xml]: Ahh, nice!  Clever feature.  No problem.  Thanks for the fix!

**Pull request merged**